h/static/scripts/directive/annotation.js

```diff
@@ -426,23 +445,26 @@ function AnnotationController(
             errorMessage(reason), 'Deleting annotation failed');
         };
         $scope.$apply(function() {
-          annotationMapper.deleteAnnotation(model).then(
+          annotationMapper.deleteAnnotation(domainModel).then(
             null, onRejected);
         });
       }
     }, true);
   };
 
+  /** Return some metadata extracted from the annotated document. */
+  vm.document = function() {
```

Comment:
Something to be aware of here is that any functions which are used in Angular expressions will be run every time the application does a full digest cycle. If this kind of function were to do something expensive, it could have a performance impact.